src/luaharfbuzz/luaharfbuzz.c

```diff
@@ -10,21 +10,20 @@ int shape_full (lua_State *L) {
   unsigned int num_features = lua_rawlen(L, 3);
   Feature *features = (Feature *) malloc (num_features * sizeof(hb_feature_t));
 
-  lua_pushnil(L);
-  while (lua_next(L, 3) != 0) {
+  for (i = 0; i != num_features; ++i) {
+    lua_geti(L, 3, i + 1);
     Feature* f = (Feature *)luaL_checkudata(L, -1, "harfbuzz.Feature");
-    features[i++] = *f;
+    features[i] = *f;
     lua_pop(L, 1);
   }
 
   const char **shapers = NULL;
   size_t num_shapers = lua_rawlen(L, 4);
   if (num_shapers) {
-    i = 0;
     shapers = (const char**) calloc (num_shapers + 1, sizeof(char*));
-    lua_pushnil(L);
-    while (lua_next(L, 4) != 0) {
-      shapers[i++] = luaL_checkstring(L, -1);
+    for (i = 0; i != num_shapers; ++i) {
```

Comment:
Same here.

Reply:
```suggestion
    for (local i = 0; i != num_shapers; ++i) {
```

Reply:
@deepakjois Actually `local` would be Lua syntax. C needs `unsigned int` here. I will push the change once my system boots again...

Reply:
Wow, that is embarassing :)! Sorry for the oversight. I will wait for you to make the changes.